inference-engine/src/transformations/src/transformations/common_optimizations/clamp_fusion.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/clamp_fusion.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ClampFusion, "ClampFusion", 0);
+
+ngraph::pass::ClampFusion::ClampFusion() {
+    auto data_pattern = ngraph::pattern::any_input();
+    auto min_const_pattern = ngraph::pattern::wrap_type<opset5::Constant>();
+    auto max_const_pattern = ngraph::pattern::wrap_type<opset5::Constant>();
+    auto max_pattern = ngraph::pattern::wrap_type<opset5::Maximum>({data_pattern, min_const_pattern});
```

Comment:
Probably it doesn't break matching patterns, but shouldn't `max_pattern` take `max_const_pattern` as argument? (the same for `min_pattern`)

Reply:
no, that's not the way clamp behaves. Clamp does `minimum(maximum(input, min_val), max_val)`

Reply:
^^ right it makes sense ;)